tests/testsuite/pgo.rs

```diff
@@ -0,0 +1,111 @@
+//! Test if PGO works.
+
+use std::path::PathBuf;
+use std::process::Command;
+
+use cargo_test_support::prelude::*;
+use cargo_test_support::project;
+use cargo_test_support::str;
+
+fn llvm_profdata() -> Option<PathBuf> {
+    let output = Command::new("rustc")
+        .arg("--print=target-libdir")
+        .output()
+        .expect("rustc to run");
+    assert!(output.status.success());
+    let mut libdir = PathBuf::from(String::from_utf8(output.stdout).unwrap());
+    assert!(libdir.pop());
+    let mut bin = libdir.join("bin").join("llvm-profdata");
+    bin.exists().then(|| bin.clone()).or_else(|| {
+        bin.set_extension("exe");
+        bin.exists().then_some(bin)
+    })
+}
+
+#[cargo_test]
+fn pgo_works() {
+    if cfg!(not(target_os = "linux")) {
+        // macOS may emit different LLVM PGO warnings.
+        // Windows LLVM has different requirements.
+        return;
+    }
```

Comment:
Can you make sure to use the `ignore` attribute when disabling a test? Otherwise it can be confusing when tests don't seem to change when you expect them to, are say that they are passing when you expect them to fail.

I realize there's still a big hole when llvm-profdata is missing. We could maybe add that to our requires mechanism?

```suggestion
#[cargo_test]
// macOS may emit different LLVM PGO warnings.
// Windows LLVM has different requirements.
#[cfg_attr(not(target_os = "linux"), ignore = "linux only")]
fn pgo_works() {
```